packages/web-client/tests/acceptance/create-merchant-test.ts

```diff
@@ -322,4 +352,74 @@ module('Acceptance | create merchant', function (hooks) {
       );
     });
   });
+
+  test('It cancels the workflow if there are no prepaid cards associated with the EOA', async function (assert) {
+    let layer2Service = this.owner.lookup('service:layer2-network')
+      .strategy as Layer2TestWeb3Strategy;
+    layer2Service.test__simulateAccountsChanged([layer2AccountAddress]);
+    layer2Service.test__simulateBalances({
+      defaultToken: new BN(0),
+    });
+
+    layer2Service.test__simulateAccountSafes(layer2AccountAddress, []);
+
+    await visit('/card-pay/merchant-services?flow=create-merchant');
+    assert
+      .dom(
+        '[data-test-postable] [data-test-layer-2-wallet-card] [data-test-address-field]'
+      )
+      .containsText(layer2AccountAddress)
+      .isVisible();
+
+    await settled();
+
+    assert
+      .dom('[data-test-postable="0"][data-test-cancelation]')
+      .containsText(
+        `It looks like you don’t have a prepaid card in your wallet. You will need one to pay the ${MERCHANT_CREATION_FEE_IN_SPEND} SPEND (${formatUsd(
+          spendToUsd(MERCHANT_CREATION_FEE_IN_SPEND)!
+        )}) merchant creation fee. Please buy a prepaid card in your Card Wallet mobile app before you continue with this workflow.`
+      );
+    assert
+      .dom('[data-test-workflow-default-cancelation-cta="create-merchant"]')
+      .containsText('Workflow canceled');
+  });
+
+  test('It cancels the workflow if prepaid cards associated with the EOA do not have enough balance', async function (assert) {
+    let layer2Service = this.owner.lookup('service:layer2-network')
+      .strategy as Layer2TestWeb3Strategy;
+    layer2Service.test__simulateAccountsChanged([layer2AccountAddress]);
+    layer2Service.test__simulateBalances({
+      defaultToken: new BN(0),
+    });
+
+    layer2Service.test__simulateAccountSafes(layer2AccountAddress, [
+      createMockPrepaidCard(
+        layer2AccountAddress,
+        prepaidCardAddress,
+        MERCHANT_CREATION_FEE_IN_SPEND - 1
+      ),
+    ]);
+
+    await visit('/card-pay/merchant-services?flow=create-merchant');
+    assert
+      .dom(
+        '[data-test-postable] [data-test-layer-2-wallet-card] [data-test-address-field]'
+      )
+      .containsText(layer2AccountAddress)
+      .isVisible();
```

Comment:
Something containing text doesn't necessarily mean the text is visible, text can be visually hidden because it is screenreader-specific. I think the person who wrote this wrote it with that in mind?

Reply:
I guess the other thing it tests for is that the workflow is scrolling down and keeping the latest item in view.